File: LAB6.cpp
#include <iostream>
#include <cmath>

using namespace std;

double sum_series(double n, int x) {
    if (x == 0) {
        return 1;
    }
    double term = pow(n, x) / tgamma(x + 1);
    return term + sum_series(n, x - 1);
}

//sum_series function overloaded.
double sum_series() {
    double n;
    cout << "Enter the value of n: ";
    cin >> n;

    int x;
    cout << "Enter the value of x: ";
    cin >> x;

    if (x == 0) {
        return 1;
    }

    double term = pow(n, x) / tgamma(x + 1);
    return term + sum_series(n, x - 1);
}

int main() {
    double n;
    int x;
    cout << "Enter the value of n: ";
    cin >> n;
    cout << "Enter the value of x: ";
   cin >> x;
    double result1 = sum_series(n, x);
    cout << "Result is : e^" << n << " = " << result1 <<endl;

    double result2 = sum_series();
    cout << "Result is : e^n = " << result2 <<endl;
    return 0;
}

File: LAB6.py
from math import factorial

n = float(input("Enter the value of n: \n"))
x = int(input("Enter the value of x: \n"))

series = lambda n, x: [n ** i / factorial(i) for i in range(x + 1)]
sum_series = lambda series: sum(series)

result = sum_series(series(n, x))

print("Result is: "f"e^{n} = {result:.3f}")


def sum(n):
    """
    Recursively calculate the sum of series (-1)^(k+1)/k from k = 1 to k = n.
    Global variable 'total'.

    :param n(int): The number of terms to sum
    :return: None
    """
    global total

    if n == 0:
        return
    term = (-1) ** (n + 1) / n
    total += term

    sum(n - 1)


total = 0
sum(6)
print(f"The sum is {total:.4f}")
